client/src/pages/lessons/edit.tsx

```diff
@@ -96,48 +103,44 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+const newLesson = {
+  lessonId: uuid(),
+  name: "Display name for the lesson",
+  intro:
+    "Introduction to the lesson,  e.g. 'This is a lesson about RGB colors'",
+  question:
+    "Question the student needs to answer, e.g. 'What are the colors in RGB?'",
+  conclusion: [
+    "Add a conclusion statement, e.g. 'In summary,  RGB colors are red, green, and blue'",
+  ],
+  expectations: [
+    {
+      expectation: "Add a short ideal answer for an expectation, e.g. 'Red'",
+      hints: [
+        {
+          text:
+            "Add a hint to help for the expectation, e.g. 'One of them starts with R'",
+        },
+      ],
+      features: null,
+    },
+  ],
+};
+
 const LessonEdit = (props: {
   search: { lessonId: string; trainStatusPollInterval?: number };
 }) => {
   const { lessonId } = props.search;
+  const [cookies] = useCookies(["accessToken"]);
+  const context = useContext(ToggleContext);
   const trainStatusPollInterval = !isNaN(
     Number(props.search.trainStatusPollInterval)
   )
     ? Number(props.search.trainStatusPollInterval)
     : TRAIN_STATUS_POLL_INTERVAL_DEFAULT;
   const classes = useStyles();
-  const [cookies] = useCookies(["user"]);
-  const [change, setChange] = React.useState(false);
-  const newLesson = {
-    lessonId: uuid(),
-    createdBy: cookies.user || "",
-    name: "Display name for the lesson",
-    intro:
-      "Introduction to the lesson,  e.g. 'This is a lesson about RGB colors'",
-    question:
-      "Question the student needs to answer, e.g. 'What are the colors in RGB?'",
-    image: "",
-    conclusion: [
-      "Add a conclusion statement, e.g. 'In summary,  RGB colors are red, green, and blue'",
-    ],
-    expectations: [
-      {
-        expectation: "Add a short ideal answer for an expectation, e.g. 'Red'",
-        hints: [
-          {
-            text:
-              "Add a hint to help for the expectation, e.g. 'One of them starts with R'",
-          },
-        ],
-        features: null,
-      },
-    ],
-    features: null,
-    isTrainable: false,
-    lastTrainedAt: "",
-  };
-  const [lesson, setLesson] = React.useState(newLesson);
-  const [loaded, setLoaded] = React.useState(false);
+  const [lesson, setLesson] = React.useState<any>();
```

Comment:
if you look below there are a whole bunch of sequences `setEditted`, `setLesson` calls below.

Generally when you see that, look for a way to combine the items of state that are inherently connected into a single state item. This is both for better adherence to the DRY principal and also (in the React case) because each individual call to a `setSTATE` hook triggers a render, and really what you want is a single render that has both aspects of the state updated.

So I recommend solve like this:

```typescript
interface LessonUnderEdit {
  lesson?: Lesson
  dirty?: boolean
}
const [lessonUnderEdit, setLessonUnderEdit] = React.useState<LessonUnderEdit>({});
```

...and then all the places where there are the sequence of set calls below replace with

```typescript
setLessonUnderEdit({
  lesson,
  dirty: true
})
```